doc/new-guidelines.md

```diff
@@ -784,6 +784,24 @@ src
         └── view.cljs
 ```
 
+### Deprecation process
+
+To properly indicate that a function is deprecated, add the ^:deprecated metadata to it. This not only formally marks the function as deprecated but also enables us to utilize linting tools to prevent its use in new code. One of the key benefits of this approach is the ability to include inline comments guiding developers to the recommended alternative function.
+
+Example:
+
+Here's an illustrative example in Clojure demonstrating how to add ^:deprecated metadata along with a comment directing to the alternative function:
+
+```clojure
+(defn window
+  {:deprecated "Use <xyz> instead"}
+  ([]
```

Comment:
To keep things simple in the guidelines, the example code could be simplified, because the actual function body doesn't matter.

Suggestion:

```clojure
(defn foo
  {:deprecated "Use some.namespace/var-name instead."}
  []
  (bar))
```